compiler/packages/babel-plugin-react-compiler/src/HIR/BuildHIR.ts

```diff
@@ -2113,90 +2113,24 @@ function lowerExpression(
-          CompilerError.throwTodo({
-            reason: `Support <${tagName}> tags where '${tagName}' is a local variable instead of a global`,
-            loc: openingIdentifier.node.loc ?? GeneratedSource,
-            description: null,
-            suggestions: null,
-          });
-        }
-        // see `error.todo-multiple-fbt-plural` fixture for explanation
-        const fbtLocations = {
-          enum: new Array<SourceLocation>(),
-          plural: new Array<SourceLocation>(),
-          pronoun: new Array<SourceLocation>(),
-        };
-        expr.traverse({
-          JSXClosingElement(path) {
-            path.skip();
-          },
-          JSXNamespacedName(path) {
-            if (path.node.namespace.name === tagName) {
-              switch (path.node.name.name) {
-                case 'enum':
-                  fbtLocations.enum.push(path.node.loc ?? GeneratedSource);
-                  break;
-                case 'plural':
-                  fbtLocations.plural.push(path.node.loc ?? GeneratedSource);
-                  break;
-                case 'pronoun':
-                  fbtLocations.pronoun.push(path.node.loc ?? GeneratedSource);
-                  break;
-              }
-            }
-          },
-        });
-        for (const [name, locations] of Object.entries(fbtLocations)) {
-          if (locations.length > 1) {
-            CompilerError.throwTodo({
-              reason: `Support <${tagName}> tags with multiple <${tagName}:${name}> values`,
-              loc: locations.at(-1) ?? GeneratedSource,
-              description: null,
-              suggestions: null,
-            });
-          }
-        }
-      }
 
-      let children: Array<Place>;
-      if (
-        tag.kind === 'BuiltinTag' &&
-        (tag.name === 'fbt' || tag.name === 'fbs')
-      ) {
-        children = expr
-          .get('children')
-          .map(child => {
-            if (child.isJSXText()) {
-              /*
-               * FBT whitespace normalization differs from standard JSX:
-               * https://github.com/facebook/fbt/blob/0b4e0d13c30bffd0daa2a75715d606e3587b4e40/packages/babel-plugin-fbt/src/FbtUtil.js#L76-L87
-               */
-              const text = child.node.value.replace(/[^\S\u00A0]+/g, ' ');
-              return lowerValueToTemporary(builder, {
-                kind: 'JSXText',
-                value: text,
-                loc: child.node.loc ?? GeneratedSource,
-              });
-            }
-            return lowerJsxElement(builder, child);
-          })
-          .filter(notNull);
-      } else {
-        children = expr
-          .get('children')
-          .map(child => lowerJsxElement(builder, child))
-          .filter(notNull);
+      const isFbt =
+        tag.kind === 'BuiltinTag' && (tag.name === 'fbt' || tag.name === 'fbs');
+      if (isFbt) {
+        checkFbtTodos(expr, tag, opening, builder);
```

Comment:
(just moved this code to the bottom of the file for readability, happy to revert / separate out to a new PR)